src/Microsoft.AspNet.Mvc.Common/TypeExtensions.cs

```diff
@@ -1,122 +0,0 @@
-// Copyright (c) .NET Foundation. All rights reserved.
-// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
-
-using System;
-using System.Linq;
-using System.Reflection;
-using Microsoft.Framework.Internal;
-
-namespace Microsoft.AspNet.Mvc
-{
-    internal static class TypeExtensions
-    {
-        // NOTE: Do not make #105 worse! Do not add new extension methods that conflict w/ .NET 4.5 methods. The
-        // exising NETFX_CORE || DNXCORE50 methods should go away (soon).
-#if NETFX_CORE || DNXCORE50
-        private static bool EqualTo([NotNull] this Type[] t1, [NotNull] Type[] t2)
-        {
-            if (t1.Length != t2.Length)
-            {
-                return false;
-            }
-
-            for (var idx = 0; idx < t1.Length; ++idx)
-            {
-                if (t1[idx] != t2[idx])
-                {
-                    return false;
-                }
-            }
-
-            return true;
-        }
-
-        public static ConstructorInfo GetConstructor([NotNull] this Type type, Type[] types)
-        {
-            return type.GetTypeInfo().DeclaredConstructors
-                                     .Where(c => c.IsPublic)
-                                     .SingleOrDefault(c => c.GetParameters()
-                                                            .Select(p => p.ParameterType).ToArray().EqualTo(types));
-        }
-#endif
-
-        public static Type BaseType([NotNull] this Type type)
-        {
-            return type.GetTypeInfo().BaseType;
-        }
-
-        public static Type ExtractGenericInterface([NotNull] this Type queryType, Type interfaceType)
```

Comment:
could you please leave this method as-is for a very short time? the intermediate change will just my move of this method to the Common repo more difficult.